### PR TITLE
mimir-mixin: add dashboard for experimental block-builder

### PR DESCRIPTION
**Pull request by narqo**
#### What this PR does

The PR is the first iteration of a new experimental dashboard, with basic monitoring for the block-builder. The dashboard is disabled by default. It must be enabled via the `$._config.block_builder_enabled` config in jsonnet.

Screenshot 2024-10-21 at 19 31 28

### CODE REVIEW

**Review by seizethedave — APPROVED** [review]

File: operations/mimir-mixin/dashboards/block-builder.libsonnet [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+(import 'dashboard-utils.libsonnet') +
+{
+  [filename]:
+    assert std.md5(filename) == 'c565e768420d79d0a632b3135d47cb30' : 'UID of the dashboard has changed, please update references to dashboard.';
+    ($.dashboard('Block-builder') + { uid: std.md5(filename) })
+    .addClusterSelectorTemplates()
+    .addRow(
+      $.row('Summary')
+      .addPanel(
+        $.timeseriesPanel('Overall consumption / sec') +
```
Comment (pracucci) [inline operations/mimir-mixin/dashboards/block-builder.libsonnet:12]: [nit] WDYT about being explicit here?

```suggestion
        $.timeseriesPanel('Kafka fetched records / sec') +
```

File: operations/mimir-mixin/dashboards/block-builder.libsonnet [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+        $.panelDescription(
+          'Overall consumption / sec',
+          'Overview of per-second rate of records consumed from Kafka.',
```
Comment (pracucci) [inline operations/mimir-mixin/dashboards/block-builder.libsonnet:15]: This is fetched, not necessarily successfully processed, right?

```suggestion
          'Overview of per-second rate of records fetched from Kafka.',
```

File: operations/mimir-mixin/dashboards/block-builder.libsonnet [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+          ],
+          [
+            'successful',
+            'read errors',
+          ],
+        ) +
+        $.stack,
+      )
+      .addPanel(
+        $.timeseriesPanel('Per-instance consumption / sec') +
```
Comment (pracucci) [inline operations/mimir-mixin/dashboards/block-builder.libsonnet:30]: [nit] Suggestion to keep it consistent with other dashboards:

```suggestion
        $.timeseriesPanel('Per pod Kafka fetched records / sec') +
```

File: operations/mimir-mixin/dashboards/block-builder.libsonnet [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+        $.panelDescription(
+          'Per-instance consumption / sec',
+          '',
+        ) +
+        $.queryPanel(
+          'rate(cortex_ingest_storage_reader_fetch_records_total{%(job)s}[$__rate_interval])' % { job: $.jobMatcher($._config.job_names.block_builder) },
```
Comment (pracucci) [inline operations/mimir-mixin/dashboards/block-builder.libsonnet:36]:
```suggestion
          'sum by(pod) (rate(cortex_ingest_storage_reader_fetch_records_total{%(job)s}[$__rate_interval]))' % { job: $.jobMatcher($._config.job_names.block_builder) },
```

File: operations/mimir-mixin/dashboards/block-builder.libsonnet [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+    .addRow(
+      $.row('')
+      .addPanel(
+        $.timeseriesPanel('Partition processing / sec') +
+        $.panelDescription(
+          'Partition processing / sec',
+          'Per-partition rate of consumption cycles',
+        ) +
+        $.queryPanel(
+          'sum by (partition) (histogram_count(rate(cortex_blockbuilder_process_partition_duration_seconds{%(job)s}[$__rate_interval])))' % [$.jobMatcher($._config.job_names.block_builder)],
```
Comment (pracucci) [inline operations/mimir-mixin/dashboards/block-builder.libsonnet:51]: Not sure a rate is a very useful information here. We expect a cycle every several minutes, right?

File: operations/mimir-mixin/dashboards/block-builder.libsonnet [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+          '{{partition}}'
+        ) +
+        $.bars,
+      )
+      .addPanel(
+        $.timeseriesPanel('Lag records') +
+        $.panelDescription(
+          'Lag records',
```
Comment (pracucci) [inline operations/mimir-mixin/dashboards/block-builder.libsonnet:59]:
```suggestion
          'Per partition records lag',
```

File: operations/mimir-mixin/dashboards/block-builder.libsonnet [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+            Number of records in the partition's backlog, as seen when starting a consumption cycle.
+          |||
+        ) +
+        $.queryPanel(
+          'max by (partition) (cortex_blockbuilder_consumer_lag_records{%(job)s}) > 0' % [$.jobMatcher($._config.job_names.block_builder)],
+          '{{partition}}',
+        )
+      )
+      .addPanel(
+        $.timeseriesPanel('Partition processing duration') +
```
Comment (pracucci) [inline operations/mimir-mixin/dashboards/block-builder.libsonnet:70]: Can you add a panel description here too, please?

**Review by pracucci — APPROVED** [review]
Thanks! I left few suggestions.

**Pull request merged**